### PR TITLE
[#1004] clean up compiler warnings

### LINKED ISSUE #1004: Clean up compiler warnings
Currently, there around 300 warnings in the code, they should be cleaned up

### PR DESCRIPTION
**Pull request by seanpmcn**
fixes #1004 

Following this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [GitHub issue](https://github.com/apache/shiro/issues) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a GitHub issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[#XXX] - Fixes bug in SessionManager`,
       where you replace `#XXX` with the appropriate GitHub issue. Best practice
       is to use the GitHub issue title in the pull request title and in the first line of the commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] add `fixes #XXX` if merging the PR should close a related issue.
 - [x] Run `mvn verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If you have a group of commits related to the same change, please squash your commits into one and force push your branch using `git rebase -i`. 
 - [x] Committers: Make sure a milestone is set on the PR
 
Trivial changes like typos do not require a GitHub issue (javadoc, comments...). 
In this case, just format the pull request title like `[DOC] - Add javadoc in SessionManager`.
 
If this is your first contribution, you have to read the [Contribution Guidelines](https://github.com/apache/shiro/blob/master/CONTRIBUTING.md)

If your pull request is about ~20 lines of code you don't need to sign an [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) 
if you are unsure please ask on the developers list.

To make clear that you license your contribution under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by seanpmcn** [comment]
I am still new to GitHub PRs. Most of my experience is with BitBucket. How do I assign a milestone, and how do I know which milestone to use?

**Comment by lprimak** [comment]
That’s for committers. Don’t worry about that

**Comment by seanpmcn** [comment]
@lprimak 
Would I not be considered a committer?

**Comment by lprimak** [comment]
Not yet :) You will be a Contributor.

File: config/ogdl/src/main/java/org/apache/shiro/config/ogdl/CommonsInterpolator.java [hunk 1]
```diff
@@ -18,10 +18,7 @@
  */
 package org.apache.shiro.config.ogdl;
 
-import org.apache.commons.configuration2.interpol.ConfigurationInterpolator;
-import org.apache.commons.configuration2.interpol.ConstantLookup;
-import org.apache.commons.configuration2.interpol.EnvironmentLookup;
-import org.apache.commons.configuration2.interpol.SystemPropertiesLookup;
+import org.apache.commons.configuration2.interpol.*;
```
Comment (lprimak) [inline config/ogdl/src/main/java/org/apache/shiro/config/ogdl/CommonsInterpolator.java:21]: Please don't use star imports. Those are usually considered bad form. Individual imports are great.

Comment (seanpmcn) [reply config/ogdl/src/main/java/org/apache/shiro/config/ogdl/CommonsInterpolator.java:21]: Thanks for pointing that out! I didn't notice my IDE adding that.

**Comment by lprimak** [comment]
Thank you for your contribution!

**Comment by seanpmcn** [comment]
@lprimak Alright. So everything else applies?

**Comment by seanpmcn** [comment]
@lprimak If you don't mind me asking, what is the procedure for responding to PR comments. Should I just make updates, commit, and push?

**Comment by lprimak** [comment]
> If you don't mind me asking, what is the procedure for responding to PR comments. Should I just make updates, commit, and push?

yes :)

> Alright. So everything else applies?

Not sure what you mean there. The issue is to fix compiler warnings, so if you fix the compiler warning that applies then :)

Try not to do too much. I did see some class renaming. I am pretty sure that would be out of scope.

**Comment by seanpmcn** [comment]
> Try not to do too much. I did see some class renaming. I am pretty sure that would be out of scope.

Oh, I'll have to take a look at that. 

By the way, would it be expected I resolve the issue or just make progress? I know some places prefer frequent pull requests while others prefer them to be basically when the entire issue is resolved.

**Comment by lprimak** [comment]
I would say to air on the side of resolving the issue in full

**Comment by seanpmcn** [comment]
Alright. I might retract this PR then. I'm not sure I want to hanging around while I work on that

**Comment by lprimak** [comment]
I would just keep working on it. No harm there in draft

**Comment by seanpmcn** [comment]
Good point. Would i have to keep squashing or could I just keep my future commits separate?

**Comment by lprimak** [comment]
I would not worry about squashing until you are ready at the end

**Comment by seanpmcn** [comment]
Alright. Thanks for taking the time to reply to me! I find it really encouraging.

**Comment by lprimak** [comment]
No problem. Looking forward to your contributions!

**Comment by seanpmcn** [comment]
> Try not to do too much. I did see some class renaming. I am pretty sure that would be out of scope.

Could you point out where that is. I don't believe I've renamed any classes, but I'm sure I may be wrong.

**Comment by seanpmcn** [comment]
Also, it seems that subclasses of HashedCredentialMatcher have been deprecated. I believe removing the implementation of those classes is out of the scope of this issue.

**Comment by lprimak** [comment]
Just leave the questionable stuff for last

**Comment by seanpmcn** [comment]
It seems like most errors are outdated documentation that includes deprecated items. I don't think I have sufficient knowledge of the code base to update the documentation. Maybe I could publish the PR without the documentation changes. Let me know what would be the best course of action.

**Comment by lprimak** [comment]
I would concentrate on resolving unchecked casts and raw types, by making them actual generics or using `@SuppressWarnings` where appropriate

**Comment by seanpmcn** [comment]
Thank you. That's clears up what direction I should take for this.

**Comment by lprimak** [comment]
This effort is abandoned for now, Anyone willing please feel free to take it over.

**Comment by lprimak** [comment]
Superseded by #1233

**Pull request closed without merging**